### PR TITLE
Add SPV_INTEL_bindless_images preview extension

### PR DESCRIPTION
**Pull request by MrSidims**
Specification: https://github.com/intel/llvm/blob/sycl/sycl/doc/design/spirv-extensions/SPV_INTEL_bindless_images.asciidoc

Co-author: He, Wenju <wenju.he@intel.com>

### CODE REVIEW

**Comment by wenju-he** [comment]
Thank you @MrSidims for the PR. Could you please also back-port to 14,15,16,17,18 branches after this PR is landed?

**Comment by MrSidims** [comment]
@wenju-he definitely

File: test/extensions/INTEL/SPV_INTEL_bindless_images/bindless_images_generic.ll [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+; CHECK-SPIRV: Extension "SPV_INTEL_bindless_images"
+; CHECK-SPIRV-DAG: TypeVoid [[#VoidTy:]]
+; CHECK-SPIRV-DAG: TypeInt [[#Int32Ty:]] 32
+; CHECK-SPIRV-DAG: TypeInt [[#Int64Ty:]] 64
+; CHECK-SPIRV-DAG: Constant [[#Int64Ty]] [[#Const42:]] 42 0
+; CHECK-SPIRV-DAG: TypeImage [[#IntImgTy:]] [[#Int32Ty]]
+; CHECK-SPIRV-DAG: TypeSampler [[#SamplerTy:]]
+; CHECK-SPIRV: FunctionParameter [[#Int32Ty]] [[#Input:]]
+; CHECK-SPIRV: ConvertHandleToImageINTEL [[#IntImgTy]] [[#]] [[#Input]]
+; CHECK-SPIRV: ConvertHandleToSamplerINTEL [[#SamplerTy]] [[#]] [[#Const42]]
```
Comment (vmaksimo) [inline test/extensions/INTEL/SPV_INTEL_bindless_images/bindless_images_generic.ll:26]: Should we also check addressing model of the input IDs as it's mentioned in spec?

File: lib/SPIRV/libSPIRV/SPIRVInstruction.h [hunk 2]
```diff
@@ -4021,5 +4021,50 @@ _SPIRV_OP(TaskSequenceRelease, false, 2, false)
...
+                   "type\n");
+
+    SPIRVType *ResTy = this->getType();
+    SPVErrLog.checkError(
+        (ResTy->isTypeImage() && OC == internal::OpConvertHandleToImageINTEL) ||
+            (ResTy->isTypeSampler() &&
+             OC == internal::OpConvertHandleToSamplerINTEL),
+        SPIRVEC_InvalidInstruction,
+        InstName + "\nIncorrect return type of the instruction must be "
+                   "image/sampler/sampled image\n");
```
Comment (vmaksimo) [inline lib/SPIRV/libSPIRV/SPIRVInstruction.h:4060]: Spec says nothing about sampled image return type for these instructions. Is this intended?

Comment (MrSidims) [reply lib/SPIRV/libSPIRV/SPIRVInstruction.h:4060]: Oops, spoiler alert, will remove :)

**Review by vmaksimo — APPROVED** [review]

**Review by vmaksimo — COMMENTED** [review]
LGTM, but please fix all CI errors first

**Pull request merged**